packages/gateway.optimism.io/src/utils/helpers.ts

```
@@ -0,0 +1,116 @@
+
+  if (!network) console.error('unsupported network!');
+
+  const l1Data = tokenList.find(
+    (tokenData: any) =>
+      tokenData.symbol === tokenSymbol &&
+      ((tokenData.chainId === chainIds.KOVAN_L1 && network === 'kovan') ||
+        (tokenData.chainId === chainIds.MAINNET_L1 && network === 'mainnet'))
+  );
+  const l2Data = tokenList.find(
+    (tokenData: any) =>
+      tokenData.symbol === tokenSymbol &&
+      ((tokenData.chainId === chainIds.KOVAN_L2 && network === 'kovan') ||
+        (tokenData.chainId === chainIds.MAINNET_L2 && network === 'mainnet'))
+  );
+  const l1TokenAddress = ethers.utils.isAddress(l1Data?.address || '') && l1Data?.address;
+  const l2TokenAddress = ethers.utils.isAddress(l2Data?.address || '') && l2Data?.address;
+  const l1BridgeAddress =
+    ethers.utils.isAddress(l1Data?.extensions.optimismGatewayAddress || '') &&
+    l1Data?.extensions.optimismGatewayAddress;
+  const l2BridgeAddress =
+    ethers.utils.isAddress(l2Data?.extensions.optimismGatewayAddress || '') &&
+    l2Data?.extensions.optimismGatewayAddress;
+
+  if (!l1TokenAddress || !l2TokenAddress || !l1BridgeAddress || !l2BridgeAddress) {
+    throw Error('One or more of the selected asset addresses are undefined');
+  }
+  return {
+    l1BridgeAddress: l1BridgeAddress as string,
+    l2BridgeAddress: l2BridgeAddress as string,
+    l1TokenAddress: l1TokenAddress as string,
+    l2TokenAddress: l2TokenAddress as string,
+    l1Messenger:
+      network === 'kovan' ? '0x78b88FD62FBdBf67b9C5C6528CF84E9d30BB28e0' : '0x902e5fF5A99C4eC1C21bbab089fdabE32EF0A5DF',
+    l2Messenger: '0x4200000000000000000000000000000000000007',
+  };
+};
+
+export const getEtherscanPrefix = (chainId: number) => {
+  switch (chainId) {
+    case chainIds.KOVAN_L1:
+      return 'kovan.';
+    case chainIds.KOVAN_L2:
+      return 'kovan-optimistic.';
+    case chainIds.MAINNET_L2:
+      return 'optimistic.';
+    default:
+      return '';
+  }
```

Perhaps default should throw an error?

---

This is just for inserting the correct subdomain string into the etherscan URL, so default creates https://etherscan.io